### PR TITLE
DSET-1791 Port CircleCI workflows to GHA

### PR DESCRIPTION
**Pull request by dlobue**
Workflow is based on s1-kafka-connect-scalyr main workflow.

I've disabled the circleci check.

Not sure how to verify the release job without actually doing a release.

DSET-1791

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+      - name: Upload Sink Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: sink-artifact
+          path: target/components/packages/scalyr-kafka-connect-scalyr-sink-${{ steps.lookupSinkVersion.outputs.result }}.zip
+
+  publish-github-release:
+    runs-on: ubuntu-latest
+    if: startsWith(github.ref, 'refs/tags/')
```
Comment (tomaz-s1) [inline .github/workflows/main.yaml:84]: This sure is better than the nasty regex we needed to use in the circle ci workflow :D

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    outputs:
+      sinkVersion: ${{ steps.lookupSinkVersion.ouputs.result }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (tomaz-s1) [inline .github/workflows/main.yaml:24]: It looks like there are some deprecation warnings so we should update some of the uses actions.

Screenshot 2023-03-21 at 10 02 50

**Review by tomaz-s1 — APPROVED** [review]
LGTM (minus a couple of small in-line comments).

I re-ran the workflow to verify that maven caching works as expected (and it appears it does https://github.com/scalyr/kafka-connect-scalyr/actions/runs/4473764488/jobs/7868376535).

Caching docker images / layers would also be nice, but not a blocker (and I know there are some edge cases associated with it).

**Review by oliverhsu77 — APPROVED** [review]
LGTM, no other comments besides what Tomaz already mentioned.  It would be good to resolve the deprecation warnings b4 merging.

File: .github/workflows/main.yaml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+          name: sink-artifact
+          path: target/components/packages/scalyr-kafka-connect-scalyr-sink-${{ steps.lookupSinkVersion.outputs.result }}.zip
+
+  publish-github-release:
+    runs-on: ubuntu-latest
+    if: startsWith(github.ref, 'refs/tags/')
+    needs:
+      - build
+    steps:
+      - name: Download test results
```
[the remark this answers is not in the document]
Comment (dlobue) [reply .github/workflows/main.yaml]: fixed.

File: .github/workflows/main.yaml (hunk 2 above)
Comment (dlobue) [reply .github/workflows/main.yaml]: These are fixed.

**Pull request merged**